lxd/container.go

```diff
@@ -660,6 +681,37 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 		}
 	}
 
+	// Check that there are no contradicting root disk devices.
+	var profileRootDiskDevices []string
+	for _, pName := range args.Profiles {
+		_, p, err := dbProfileGet(d.db, pName)
+		if err != nil {
+			return nil, fmt.Errorf("Could not load profile '%s'.", pName)
+		}
+
+		k, v := containerGetRootDiskDevice(p.Devices)
+		if k != "" && v["pool"] == "" {
+			return nil, fmt.Errorf("A root disk device must have the \"pool\" property set.")
+		} else if k != "" && !shared.StringInSlice(k, profileRootDiskDevices) {
+			profileRootDiskDevices = append(profileRootDiskDevices, k)
+		}
+	}
+
+	k, newLocalRootDiskDevice := containerGetRootDiskDevice(args.Devices)
+	// Check whether container has a local root device with a "pool"
+	// property set.
+	if k != "" && newLocalRootDiskDevice["pool"] == "" {
+		return nil, fmt.Errorf("A root disk device must have the \"pool\" property set.")
```

Comment:
@stgraber, this part: https://github.com/lxc/lxd/pull/2903/files#diff-a8c4bafcd7965c655b000037f7d485adR224

```
	if localRootDiskDeviceKey == "" || localRootDiskDeviceKey != "" && v["pool"] == "" {
		// Request came without a root disk device or without the pool
		// property set. Try to retrieve this information from a
		// profile. (No need to check for conflicting storage pool
		// properties in the profiles. This will be handled by
		// containerCreateInternal().)
		for _, pName := range req.Profiles {
			_, p, err := dbProfileGet(d.db, pName)
			if err != nil {
				return InternalError(err)
			}

			k, v := containerGetRootDiskDevice(p.Devices)
			if k != "" && v["pool"] != "" {
				poolForCopyOrMove = v["pool"]
				break
			}
		}
		if poolForCopyOrMove == "" {
			pools, err := dbStoragePools(d.db)
			if err != nil {
				return InternalError(err)
			}

			if len(pools) != 1 {
				return InternalError(fmt.Errorf("No unique storage pool found."))
			}

			poolForCopyOrMove = pools[0]
		}
	}
```